src/core/Akka.Remote/Transport/TransportAdapters.cs

```diff
@@ -69,7 +70,7 @@ internal class TransportAdapters : IExtension
         public TransportAdapters(ExtendedActorSystem system)
         {
             System = system;
-            Settings = ((RemoteActorRefProvider)system.Provider).RemoteSettings;
+            Settings = ((IRemoteActorRefProvider)system.Provider).RemoteSettings;
```

Comment:
This is the only real change here. This call happens once at startup per `ActorSystem` the first time this `ActorSystemExtension` is used.